### ISSUE TITLE
tips about merging the master branch to Sunrise 1.7

### ISSUE THREAD

**Issue by jacknlliu** [IFL-CAMP/iiwa_stack#198]
Hi, for many reasons, we can't use the latest version of this repo. I hope to continue to maintain the Sunrise 1.7 version. Could you give me some tips about how to modify the master branch to Sunrise 1.7?

**Comment by SalvoVirga** [comment]
It shouldn't be too hard.
Try to use the current (in master) Java code in Sunrise Workbench, using the APIs from Sunrise 1.7. You will most likely have some import error. Delete the imports that give you those errors, then you are going to have errors for missing references in the code.
Just over on top of the lines whose references are missing and you should automatically get a suggestion on what to import to fix the problem.

The only thing you might need extra is the file [xml-apis-1.0.b2.jar](https://github.com/IFL-CAMP/iiwa_stack/blob/ae3ba2792048f73122edfe1abb5a3e6b25540c45/iiwa_ros_java/ROSJavaLib/xml-apis-1.0.b2.jar) that you can find in some very old commit like the one linked.

If you encounter particular errors let me know.

**Comment by GuancongLuo** [comment]
I meet the same problem. 
And I can import some package
`import com.kuka.connectivity.motionModel.smartServo.ServoMotion;`
`import com.kuka.connectivity.motionModel.smartServo.SmartServo;`
`import com.kuka.connectivity.motionModel.smartServoLIN.SmartServoLIN;`
`import com.kuka.task.ITaskLogger;`

Actually, I have the 'conectivity' package, but the package name is different with the code.

**Comment by jacknlliu** [comment]
@GuancongLuo We have updated the Sunrise OS to 1.11 using another backup hard disk, so I may not maintain the 1.7 branch recently. Good luck anyway！

**Comment by SalvoVirga** [comment]
@GuancongLuo Did you try to remove the imports that trigger the errors? Workbench should then mark the symbols with a missing import and suggest which import to add.
If that doesn't help, just explore the KUKA API's a bit, you should be able to find the right imports quite easily. You will also need to change the `TaskLogger` import btw, that change with some more modern version.

Let me know if you solved it, else I could try to install a Sunrise 1.7 on a VM

**Comment by jacknlliu** [comment]
@GuancongLuo I have committed a branch for Sunrise 1.7 [here](https://github.com/jacknlliu/iiwa_stack/tree/update-v1.7-from-upstream).

**Comment by GuancongLuo** [comment]
@jacknlliu Thank you for your branch. I successfully installed **ROSSmartServo** application. 
@SalvoVirga https://github.com/IFL-CAMP/iiwa_stack/issues/183#issuecomment-502653195
But iiwa 14 can't connect successfully with my ROS. I check the ROS IP and sunrise IP, this is right. I don't know how to solve this problem. Could you get me some advance?

![IP addr AND roscore](https://user-images.githubusercontent.com/41011340/64676388-0f6e6c80-d4a8-11e9-8d13-83f300d46aa2.png)
ROS core and Ethernet address

![RoboticsAPI](https://user-images.githubusercontent.com/41011340/64676464-39279380-d4a8-11e9-9cae-28305e8e558a.PNG)

Sunrise WorkBeach RoboticsAPI.data.XMl

![stop error](https://user-images.githubusercontent.com/41011340/64676835-fca86780-d4a8-11e9-912c-dd2a418e5334.jpg)
Always waiting. And when I stop it, I get the error.

I need your help and thanks again.

**Comment by jacknlliu** [comment]
@GuancongLuo  make sure that you are using the **X66** port, and could you ping the robot ip from the PC?

**Comment by GuancongLuo** [comment]
Yes, I can ping the robot IP from PC. And I connected the PC with **X66** port this morning. I started **roscore** first and then started **ROSSmartServo**. It is not working.

![Screenshot from 2019-09-12 19-03-09](https://user-images.githubusercontent.com/41011340/64751195-02a45400-d54e-11e9-8a97-de353959930a.png)
![stop error](https://user-images.githubusercontent.com/41011340/64751615-4a77ab00-d54f-11e9-8640-a4bf1bd7e30f.jpg)

Still, have the same problem.

![IO model influence](https://user-images.githubusercontent.com/41011340/64751647-6a0ed380-d54f-11e9-80c5-3f05173f2e7f.jpg)

Is this a **configuration** issue for **Sunrise WorkBeach**?

**Comment by jacknlliu** [comment]
It's nice to hear that it works for you.
